## Cadenas de Markov - Desafio

El desafio consiste en crear a partir de la información extraida de meteochile.cl, cadenas de Markov y simular mediante Montecarlo varios escenarios de clima.

Actividades:

1. Escoja la estacion de clima con la que usted quiera trabajar, necesita el codigo que obtener en https://climatologia.meteochile.gob.cl/application/index/menuTematicoEmas .
3. Descargue la información anual de las variables de clima que estime necesaria, por ejemplo:

In [6]:
variables = ["temperaturaMediaAnual", "temperaturaMinimaAnual"]
años = list(range(2023, 2025))
estacion = 410005

import urllib.request as ureq

urlbase = "https://climatologia.meteochile.gob.cl/application/anual"
for v in variables:
    for a in años:
        url = urlbase + "/" + v + "/" + str(estacion) + "/" + str(a)
        print(url)
        pagina = ureq.urlopen(url)
        datos = pagina.read()
        print(datos)
        break
    break

https://climatologia.meteochile.gob.cl/application/anual/temperaturaMediaAnual/410005/2023
b'<!DOCTYPE html><html lang="es">\n\n<head>\n\n    <!-- Global site tag (gtag.js) - Google Analytics -->\n    <script async src="https://www.googletagmanager.com/gtag/js?id=G-K73XP588R6"></script>\n    <script>\n    window.dataLayer = window.dataLayer || [];\n       function gtag(){dataLayer.push(arguments);}\n       gtag(\'js\', new Date());\n       gtag(\'config\', \'G-K73XP588R6\');\n    </script>\n\n    <meta charset="utf-8">\n    <meta http-equiv="Content-Security-Policy" content="">\n    <title>Servicios Clim\xc3\xa1ticos</title>    <meta name="viewport" content="width&#x3D;device-width,&#x20;initial-scale&#x3D;1.0">\n<meta http-equiv="X-UA-Compatible" content="IE&#x3D;edge">\n    <!-- Le styles -->\n    <link href="&#x2F;css&#x2F;weather-icons&#x2F;css&#x2F;weather-icons.min.css" media="screen" rel="stylesheet" type="text&#x2F;css">\n<link href="&#x2F;css&#x2F;font-awesome&#x2F;css&#x2F;fo

In [ ]:
# sacamos la info desde datos...

info = str(datos).split("<td class=\"text-center\">")
print(info[1:14])

['  \\n                     1                  </td>\\n                                       ', '  \\n                        14.1                     </td>\\n                                       ', '  \\n                        13.3                     </td>\\n                                       ', '  \\n                        14.5                     </td>\\n                                       ', '  \\n                        13.5                     </td>\\n                                       ', '  \\n                        7.1                     </td>\\n                                       ', '  \\n                        8.5                     </td>\\n                                       ', '  \\n                        11.5                     </td>\\n                                       ', '  \\n                        8.3                     </td>\\n                                       ', '  \\n                        7.0                     </td>\\n    

De arriba se desprende que la temperatura Media el 1 de enero de 2023 fue 14.1 grados, el 1 de febrero de 2023 fue 13.3 grados, y el 1 de diciembre de 2023 fue 11.5 grados, debe extraerla automaticamente y crear una estructura de datos apropiada. Luego de esto etiquete cada dia, con los estados que usted definio.

3. Cree 12 matrices de transición, con los datos historicos de 25 años (2000 al 2024), de acuerdo al cambio de estados diario, le conviene contarlos y al final dividirlo por el número de dias para estimar la probabilidad aproximada de la matriz de transición.

4. Con cada matriz de transición, simule 100 experimentos (100 años distintos) y grafique de algun modo la distribución que obtuvo de acuerdo a sus estados.

5. Con cada matriz de transición obtenga (si se puede), los estados estacionarios de las matrices.

6. Programa como se dibuja la matriz de transición con alguna libreria de grafos.

Suerte!